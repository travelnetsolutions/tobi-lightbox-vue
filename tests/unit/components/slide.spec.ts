import Slide from '@/components/slide.vue';
import { createWrapper, shallowMount } from '@vue/test-utils';
import { gallery } from './fixtures/galleryFixture';

describe('Lightbox.vue', () => {

  let props: {};

  const build = () => {
    const wrapper = shallowMount(Slide, {
      propsData: props,
    });
    return {
      wrapper,
      img: () => wrapper.find('img'),
    };
  };

  beforeEach(() => {
    props = {
      image: {},
    };
  });

  it('renders the component', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('doesn\'t renders the image if the current index isn\'t/hasn\'t been within 1 of the index', () => {
    const { wrapper, img } = build();
    const imageIndex = 5;
    const image = gallery[imageIndex];
    wrapper.setProps({
      image,
      index: imageIndex,
      currentIndex: 0,
    });
    expect(img().attributes('src')).toBeFalsy();
  });
  it('renders the image if the current index is within 1 of the index', () => {
    const { wrapper, img } = build();
    const image = gallery[0];
    wrapper.setProps({
      image,
      index: 1,
      currentIndex: 0,
    });
    expect(img().attributes('src')).toBe(image.imgsrc);
  });
  it('renders the image if it has rendered before', () => {
    const { wrapper, img } = build();
    const image = gallery[0];
    wrapper.setProps({
      image,
      index: 1,
      currentIndex: 0,
    });
    wrapper.setProps({
      currentIndex: 5,
    });
    expect(img().attributes('src')).toBe(image.imgsrc);
  });
  it('has alt text when the image is rendered', () => {
    const { wrapper, img } = build();
    const image = gallery[0];
    wrapper.setProps({
      image,
      index: 1,
      currentIndex: 0,
    });
    expect(img().attributes('alt')).toBe(image.description);
  });
});
