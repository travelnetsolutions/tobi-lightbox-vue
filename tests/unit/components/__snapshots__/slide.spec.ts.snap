// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lightbox.vue renders the component 1`] = `
<div class="tobi__slider__slide">
  <div class="tobi__slider__slide__content">
    <figure><img></figure>
  </div>
</div>
`;
